pkg/daemon/daemon.go

```diff
@@ -1367,7 +1382,7 @@ func compareOSImageURL(current, desired string) bool {
 // Otherwise if `false` is returned, then we need to perform an update.
 func (dn *Daemon) checkOS(osImageURL string) bool {
 	// Nothing to do if we're not on RHCOS or FCOS
-	if dn.OperatingSystem != MachineConfigDaemonOSRHCOS && dn.OperatingSystem != MachineConfigDaemonOSFCOS {
+	if !dn.isOstree {
```

Comment:
Avoiding the need to mentally apply a [DeMorgan rule](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) when reading this code is IMO a clear win.